adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java

```diff
@@ -114,4 +115,46 @@ void getRouteByIdShouldReturnRelatedRoute() {
         assertEquals(mockRoute00, underTest.getRouteById("route id0"));
         assertEquals(mockRoute01, underTest.getRouteById("route id1"));
     }
+
+    @Test
+    public void callToAddTripShouldAddTripToRepoAndReturnSameEntity() {
+        final Trip mockTrip = mock(Trip.class);
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        when(mockTrip.getTripId()).thenReturn("trip id");
+
+        assertEquals(underTest.addTrip(mockTrip), mockTrip);
+    }
+
+    @Test
+    public void addSameTripTwiceShouldReturnNull() {
+        final Trip mockTrip = mock(Trip.class);
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        when(mockTrip.getTripId()).thenReturn("trip id");
+
+        underTest.addTrip(mockTrip);
+
+        assertNull(underTest.addTrip(mockTrip));
+    }
+
+    @Test
+    void addNullTripShouldThrowIllegalArgumentException() {
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        assertThrows(IllegalArgumentException.class, () -> underTest.addTrip(null));
+    }
+
+    @Test
+    public void getTripByIdShouldReturnRelatedTrip() {
```

Comment:
Ok I see the add is tested here. I suggest the first test method and that one be refactored as one.